cupyx/scipy/interpolate/_bspline.py

```diff
@@ -0,0 +1,934 @@
+
+import operator
+
+import cupy
+from cupy._core import internal
+from cupy._core._scalar import get_typename
+
+from cupyx.scipy.sparse import csr_matrix
+
+import numpy as np
+
+TYPES = ['double', 'complex<double>']
+INT_TYPES = ['int', 'long long']
+
+INTERVAL_KERNEL = r'''
+#include <cupy/complex.cuh>
+extern "C" {
+__global__ void find_interval(
+        const double* t, const double* x, long long* out,
+        int k, int n, bool extrapolate, int total_x) {
+
+    int idx = blockDim.x * blockIdx.x + threadIdx.x;
+    if(idx >= total_x) {
+        return;
+    }
+
+    double xp = *&x[idx];
+    double tb = *&t[k];
+    double te = *&t[n];
+
+    if(isnan(xp)) {
+        out[idx] = -1;
+        return;
+    }
+
+    if((xp < tb || xp > te) && !extrapolate) {
+        out[idx] = -1;
+        return;
+    }
+
+    int left = k;
+    int right = n;
+    int mid;
+    bool found = false;
+
+    while(left < right && !found) {
+        mid = ((right + left) / 2);
+        if(xp > *&t[mid]) {
+            left = mid + 1;
+        } else if (xp < *&t[mid]) {
+            right = mid - 1;
+        } else {
+            found = true;
+        }
+    }
+
+    int default_value = left - 1 < k ? k : left - 1;
+    int result = found ? mid + 1 : default_value + 1;
+
+    while(xp >= *&t[result] && result != n) {
+        result++;
+    }
+
+    out[idx] = result - 1;
+}
+}
+'''
+
+INTERVAL_MODULE = cupy.RawModule(
+    code=INTERVAL_KERNEL, options=('-std=c++11',),)
+#    name_expressions=[f'find_interval<{type_name}>' for type_name in TYPES])
+
+
+D_BOOR_KERNEL = r'''
+#include <cupy/complex.cuh>
+#include <math_constants.h>
```

Comment:
Addresses Jenkins failures, this is from ROCm environment.

```suggestion
#include <cupy/math_constants.h>
```